### PR TITLE
fixes penetrator rounds phasing through vehicles

### PR DESCRIPTION
**Pull request by InvalidArgument3**
## About The Pull Request

When I gave vehicles the PASSMACHINE flag in #61793 I forgot that penetrator rounds existed leading to them completely ignoring mechs. This time I created a new flag for vehicles, added it to water particle effects, and excluded it from the penetrator rounds. 

Additionally this flag has been added to the heretic's phasing ability.

Please add the GBP: no update tag since this was my fault.

## Why It's Good For The Game

lazily copy pasting flags bad. Making new flags and implementing properly good.
## Changelog

:cl:
fix: penetrator rounds will no longer ignore vehicles
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
make that heretic phasing thing still phase through vehicles

**Comment by InvalidArgument3** [comment]
> make that heretic phasing thing still phase through vehicles

done

File: code/datums/status_effects/buffs.dm [hunk 1]
```diff
@@ -371,13 +371,13 @@
 	. = ..()
 	to_chat(owner,span_notice("You phase through reality, nothing is out of bounds!"))
 	owner.alpha = 180
-	owner.pass_flags |= PASSCLOSEDTURF | PASSGLASS | PASSGRILLE | PASSMACHINE | PASSSTRUCTURE | PASSTABLE | PASSMOB | PASSDOORS
+	owner.pass_flags |= PASSCLOSEDTURF | PASSGLASS | PASSGRILLE | PASSMACHINE | PASSSTRUCTURE | PASSTABLE | PASSMOB | PASSDOORS | PASSVEHICLE
```
Comment (Watermelon914) [inline code/datums/status_effects/buffs.dm:374]: include in the description of your PR too

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/datums/status_effects/buffs.dm (hunk 1 above)
Comment (InvalidArgument3) [reply code/datums/status_effects/buffs.dm:374]: alright, thanks

**Pull request merged**